polyply/src/nonbond_engine.py

```diff
@@ -318,7 +319,24 @@ def from_topology(cls, molecules, topology, box):
         for molecule in molecules:
             for node in molecule.nodes:
                 if "position" in molecule.nodes[node]:
-                    positions[idx, :] = molecule.nodes[node]["position"]
+                    # check if position is inside grid
+                    is_finite = np.all(np.isfinite(molecule.nodes[node]["position"]))
+                    if not_exceeds_max_dimensions(molecule.nodes[node]["position"], box) and is_finite:
+                        positions[idx, :] = molecule.nodes[node]["position"]
+                    elif not is_finite:
+                        mol_name = molecule.mol_name
+                        resname = molecule.nodes[node]['resname']
+                        msg = (f"Provided coordinate for residue {resname} in molecule {mol_name}\n"
+                               f"with molecule index {idx} is not finite. All coordinates in\n"
+                                "polyply must be positive semi-definite.")
+                        raise IOError(msg)
+                    else:
+                        mol_name = molecule.mol_name
+                        resname = molecule.nodes[node]['resname']
+                        msg = (f"Provided coordinate for residue {resname} in molecule {mol_name}\n"
+                               f"with molecule index {idx} exceeds maximum box dimensions.\n"
+                                "Make sure all coordiantes are wrapped inside the box.")
```

Comment:
I personally wouldn't put the \n in the messages. Let thelogging module figure out formatting

Reply:
well my experience is that the logger just dumps it into one line, which users find hard to decipher. A bit of line wrapping goes a long way.

Reply:
Mjeh. The cleanest solution would be to make/adapt a logging formatter.